### PR TITLE
[18.0][MIG] account_journal_restrict_mode: Migration to 18.0

### PR DESCRIPTION
**Pull request by SaraDForgeFlow**
Standard migration

### CODE REVIEW

**Comment by AaronHForgeFlow** [comment]
LGTM now that https://github.com/odoo/odoo/commit/b48e66ed3a4157ca65bdd4c519b797cc6509b9a3 simplifies the code

File: account_journal_restrict_mode/models/account_journal.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    _inherit = "account.journal"
+
+    restrict_mode_hash_table = fields.Boolean(default=True, readonly=True)
+
+    @api.constrains("restrict_mode_hash_table")
+    def _check_journal_restrict_mode(self):
+        for rec in self:
+            if not rec.restrict_mode_hash_table:
+                raise UserError(
+                    self.env._("Journal %s must have Lock Posted Entries enabled.") % rec.name
```
Comment (JordiMForgeFlow) [inline account_journal_restrict_mode/models/account_journal.py:18]:
```suggestion
                    _("Journal %s must have Lock Posted Entries enabled.") % rec.name
```

Comment (JordiMForgeFlow) [inline account_journal_restrict_mode/models/account_journal.py]: @MiquelRForgeFlow wow, did not see that coming 😄 thanks!

**Review by JordiMForgeFlow — APPROVED** [review]
code review

**Review by AaronHForgeFlow — COMMENTED** [review]
LGTM when the pre-commit check is passed

**Review by AaronHForgeFlow — APPROVED** [review]

**Pull request merged**